fe/fe-core/src/main/java/org/apache/doris/qe/cache/CacheAnalyzer.java

```diff
@@ -417,6 +384,51 @@ private CacheMode innerCheckCacheModeForNereids(long now) {
         return CacheMode.None;
     }
 
+    private List<CacheTable> buildCacheTableList() {
+        //Check the last version time of the table
+        MetricRepo.COUNTER_QUERY_TABLE.increase(1L);
+        long olapScanNodeSize = scanNodes.stream().filter(node -> node instanceof OlapScanNode).count();
+        long hiveScanNodeSize = scanNodes.stream().filter(node -> node instanceof HiveScanNode).count();
+        if (olapScanNodeSize > 0) {
+            MetricRepo.COUNTER_QUERY_OLAP_TABLE.increase(1L);
+        }
+        if (hiveScanNodeSize > 0) {
+            MetricRepo.COUNTER_QUERY_HIVE_TABLE.increase(1L);
+        }
+
+        if (!(olapScanNodeSize == scanNodes.size() || hiveScanNodeSize == scanNodes.size())) {
+            LOG.debug("only support olap/hive table with non-federated query, other types are not supported now, "
+                    + "queryId {}", DebugUtil.printId(queryId));
+            return Collections.emptyList();
+        }
+
+        List<CacheTable> tblTimeList = Lists.newArrayList();
+        for (int i = 0; i < scanNodes.size(); i++) {
+            ScanNode node = scanNodes.get(i);
+            if (enablePartitionCache()
+                    && (node instanceof OlapScanNode)
+                    && ((OlapScanNode) node).getSelectedPartitionNum() > 1
+                    && selectStmt != null
+                    && selectStmt.hasGroupByClause()) {
+                LOG.debug("more than one partition scanned when qeury has agg, partition cache cannot use, queryid {}",
+                        DebugUtil.printId(queryId));
+                return Collections.emptyList();
+            }
+            CacheTable cTable = node instanceof OlapScanNode
+                    ? buildCacheTableForOlapScanNode((OlapScanNode) node)
+                    : buildCacheTableForHiveScanNode((HiveScanNode) node);
+            tblTimeList.add(cTable);
+        }
+        if (olapScanNodeSize > 0) {
```

Comment:
Already add it before?

Reply:
I've checked my code again, and i do not find any other place which added `cTable` before, just here.

Reply:
Do you mean the metrics? It has been fixed.